dot/core/consensus_test.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+// TODO
```

Comment:
we need `NextEpochDescriptor` to be included in first block before we can test, maybe we should add this to #662

Reply:
NextEpochDescriptor is implemented here https://github.com/ChainSafe/gossamer/blob/f4d80d67b7e84c73a8a2213297516c3050cfd9ef/lib/babe/types.go#L150 so I think you could write a test with it?

Reply:
`NextEpochDesciptor` is #662, so it might make more sense to hold off here and have a babe session running that includes all the necessary data in the first block before testing here.

Reply:
I was having trouble mocking the babe session before as well, which is why I am hoping it will be easier to write a test once #662 is implemented but maybe I was overcomplicating it